operations/helm/charts/mimir-distributed/templates/query-scheduler/query-scheduler-dep.yaml

```
@@ -0,0 +1,132 @@
+      containers:
+        - name: query-scheduler
+          image: "{{ include "mimir.imageReference" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - "-target=query-scheduler"
+            - "-config.expand-env=true"
+            - "-config.file=/etc/mimir/mimir.yaml"
+          {{- range $key, $value := .Values.query_scheduler.extraArgs }}
+            - "-{{ $key }}={{ $value }}"
+          {{- end }}
+          volumeMounts:
+            {{- if .Values.query_scheduler.extraVolumeMounts }}
+              {{ toYaml .Values.query_scheduler.extraVolumeMounts | nindent 12}}
+            {{- end }}
+            - name: runtime-config
+              mountPath: /var/{{ include "mimir.name" . }}
+            {{- if .Values.enterprise.enabled }}
+            - name: license
+              mountPath: /license
+            {{- end }}
+            - name: config
+              mountPath: /etc/mimir
+            - name: storage
+              mountPath: /data
+              {{- if .Values.query_scheduler.persistence.subPath }}
+              subPath: {{ .Values.query_scheduler.persistence.subPath }}
+              {{- end }}
+          ports:
+            - name: http-metrics
+              containerPort: {{ include "mimir.serverHttpListenPort" . }}
+              protocol: TCP
+            - name: grpc
+              containerPort: {{ include "mimir.serverGrpcListenPort" . }}
+              protocol: TCP
+          livenessProbe:
+            {{- toYaml .Values.query_scheduler.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml .Values.query_scheduler.readinessProbe | nindent 12 }}
+          resources:
+            {{- toYaml .Values.query_scheduler.resources | nindent 12 }}
+          securityContext:
+            readOnlyRootFilesystem: true
+          env:
+            {{- with .Values.global.extraEnv }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with .Values.query_scheduler.env }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- with .Values.global.extraEnvFrom }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with .Values.query_scheduler.extraEnvFrom }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+{{- if .Values.query_scheduler.extraContainers }}
+{{ toYaml .Values.query_scheduler.extraContainers | indent 8}}
+{{- end }}
+      nodeSelector:
+        {{- toYaml .Values.query_scheduler.nodeSelector | nindent 8 }}
+      affinity:
+        {{- toYaml .Values.query_scheduler.affinity | nindent 8 }}
+      tolerations:
+        {{- toYaml .Values.query_scheduler.tolerations | nindent 8 }}
+      terminationGracePeriodSeconds: {{ .Values.query_scheduler.terminationGracePeriodSeconds }}
+      volumes:
+        - name: config
+          secret:
+            secretName: {{ tpl .Values.externalConfigSecretName . }}
```

You may need to rebase onto main for this to work. But basically it can be a secret or a configmap now and there's a handy named template that knows which one it should be.

```suggestion
        - name: config
          {{- include "mimir.configVolume" . | nindent 10 }}
```